registries/adapters/apiv2_adapter.go

```diff
@@ -0,0 +1,220 @@
+//
+// Copyright (c) 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package adapters
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/automationbroker/bundle-lib/bundle"
+	"github.com/automationbroker/bundle-lib/registries/adapters/oauth"
+	log "github.com/sirupsen/logrus"
+)
+
+const apiV2ManifestURL = "%v/v2/%v/manifests/%v"
+const apiV2CatalogURL = "%v/v2/_catalog"
+
+// NewApiV2Adapter - creates and returns a *ApiV2Adapter ready to use.
+func NewApiV2Adapter(config Configuration) *ApiV2Adapter {
+	return &ApiV2Adapter{
+		Config: config,
+		client: oauth.NewClient(config.User, config.Pass, config.SkipVerifyTLS, config.URL),
+	}
+}
+
+// ApiV2Adapter - API V2 Adapter
+type ApiV2Adapter struct {
+	Config Configuration
+	client *oauth.Client
+}
+
+// ApiV2CatalogResponse - Catalog Response
+type ApiV2CatalogResponse struct {
+	Repositories []string `json:"repositories"`
+}
+
+// RegistryName - Retrieve the registry name
+func (r ApiV2Adapter) RegistryName() string {
+	if r.Config.URL.Host == "" {
+		return r.Config.URL.Path
+	}
+	return r.Config.URL.Host
+}
+
+// GetImageNames - retrieve the images
+func (r ApiV2Adapter) GetImageNames() ([]string, error) {
+	log.Debugf("%s - GetImageNames", r.Config.AdapterName)
+	log.Debugf("BundleSpecLabel: %s", BundleSpecLabel)
+
+	// Authorization
+	err := r.client.Getv2()
```

Comment:
We need to initialize this before any call to the registry. I am wondering if this should maybe be don in the new function. This would make it so that a user does not have to call GetImagesNames before anything else?